miri

```diff
@@ -128,6 +128,26 @@ find_sysroot() {
     fi
 }
 
+# Run rustfmt on all tests and crates
+fmt() {
+    find "$MIRIDIR" -not \( -name target -prune \) -name '*.rs' \
+        | xargs rustfmt +$TOOLCHAIN --edition=2021 --config-path "$MIRIDIR/rustfmt.toml" "$@"
+}
+
+# Run clippy on all crates
+clippy() {
+    $CARGO clippy $CARGO_EXTRA_FLAGS --manifest-path "$MIRIDIR"/Cargo.toml --all-targets "$@"
+    $CARGO clippy $CARGO_EXTRA_FLAGS --manifest-path "$MIRIDIR"/ui_test/Cargo.toml --all-targets "$@"
+    $CARGO clippy $CARGO_EXTRA_FLAGS --manifest-path "$MIRIDIR"/cargo-miri/Cargo.toml "$@"
```

Comment:
You can avoid this refactoring by actually calling `$0` recursively, with some environment variable to avoid endless loops.

Reply:
idk... that seems more roundabout/fragile to me, but if you prefer, I can do that.